afl-fuzz.c

```diff
@@ -1332,29 +1327,48 @@ static void cull_queue(void) {
 
   while (q) {
     q->favored = 0;
+    q->rand = UR(INT_MAX);
```

Comment:
@jiradeto I just realized that the weighted random should be used here instead of where it is now... Something like this:

```C
double weight = 1.0;
      int r = 0;
      int rid = INT_MAX;
      while (weight >= 1.0) {
        r = UR(INT_MAX);
        if (r < rid)
          rid = r;
        weight -= 1.0;
      }
      if (weight > 0.0 && weight > rand_double()) {
        r = UR(INT_MAX);
        if (r < rid)
          rid = r;
      }
      q->rand = rid;
```

Reply:
Oh I see, the weighted random is assigned to each seed. The above issue is fixed now.

Reply:
Great! Thanks!